### ISSUE TITLE
Migrate ome_model Python package to its own repository

### ISSUE THREAD

**Issue by sbesson** [ome/ome-model#194]
The code under https://github.com/ome/ome-model/tree/v6.3.6/ome_model has had little maintenance/development over the last 4 years. 
Functionally, its content is also very different from the rest of this code base which contains:

1- the source XSD files for the OME data model
2- sample XML files for the different schema versions
3- the code-generated Java implementation of the OME model objects

Following the extraction to a separate repository, a decision should likely be made with regard to maintaining it vs archiving it in favor of other community Python projects like https://pypi.org/project/ome-types/. 

_Originally posted by @sbesson in https://github.com/ome/ome-model/pull/193#pullrequestreview-2457871918_

**Comment by joshmoore** [comment]
Especially if we feel comfortable that our existing scripts could be re-written on top of ome-types, I'd be :+1: trying to unify on it.

**Comment by sbesson** [comment]
Agreed, as a proof of concept, I worked on a minimal example upgrading a IDR script generating an HCS companion file using this package to `ome-types`.

The code difference is very reasonable - see https://github.com/IDR/idr0092-ostrop-organoid/pull/2 and the generated OME-XML is consistent with the original one with the only differences being related to attribute ordering and UUIDs.

Based on this minimal investigation, I am further inclined to remove this project and deprecate it in favor of `ome-types`. At the API level, I cannot see anything that I would care about maintaining. If any was raised as valuable, I would propose to discuss how to open it as a contribution against `ome-types` directly. Any valuable experience gained from this project could be also be converted into documentation https://ome-types.readthedocs.io/.

**Comment by sbesson** [comment]
#199 was opened to deprecate the Python code and redirect possible consumers towards using `ome-types`.
The remaining outstanding question is for @tlambert03. Summarizing briefly, https://github.com/ome/ome-model/issues/194#issuecomment-2535308325 describes using `ome-types` directly to generate OME-XML companion files decorating plain TIFF files and turning them into rich OME-TIFF filesets. Would this use case be valuable e.g. as a section in the ome-types documentation?

**Comment by tlambert03** [comment]
Hey @sbesson @joshmoore, thanks for the ping.  Sure, I would happily extend the ome-types documentation.  The use case of turning a regular tiff into an OME-TIFF is a common one that we definitely should have an example of.  Would you have time to contribute that?  (also ok if not)
